plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/completion/DAICompletionSession.java

```diff
@@ -16,35 +16,52 @@
  */
 package org.jkiss.dbeaver.model.ai.completion;
 
-import org.jkiss.code.NotNull;
-
 import java.util.ArrayList;
 import java.util.List;
 
+import org.jkiss.code.NotNull;
+
 /**
  * Completion session
  */
 public class DAICompletionSession {
-    private final List<DAICompletionMessage> messages;
+    private final List<DAICompletionMessage> allMessages;
```

Comment:
Let's store all messages in the session. It's the responsibility of the engine to collect and filter messages in which it's interested.

Please revert all changes in this file.

Reply:
Well, most chat engines will have a similar behavior regarding answers from AI, so this may lead to the code duplication

Reply:
No? All implementations inherit from the `AbstractAICompletionEngine` class, and that's where you want to filter messages.